src/Utils/utils.ts

```diff
@@ -324,3 +325,22 @@ export function getWeeklyIntervalsFromTodayTill(pastDate?: Date | string) {
 
   return intervals;
 }
+
+/**
+ * Parses a selectedSlot param (URL-encoded JSON string) into a TokenSlot object.
+ * Returns undefined if parsing fails.
+ * @param selectedSlotParam The URL param value (string or undefined)
+ * @param onError Optional error callback (e.g., for toast)
+ */
+export function parseSelectedSlotParam(
+  selectedSlotParam: string | undefined,
+  onError?: (error: unknown) => void,
+): TokenSlot | undefined {
+  if (!selectedSlotParam) return undefined;
+  try {
+    return JSON.parse(decodeURIComponent(selectedSlotParam));
+  } catch (error) {
+    if (onError) onError(error);
+    return undefined;
+  }
+}
```

Comment:
i don't see any reason why this is a globally declared utility fn. 

this can be scoped to schedule's utility fn's file